Lib/test/test_ctypes/test_parameters.py

```diff
@@ -224,6 +225,8 @@ def __dict__(self):
         with self.assertRaises(ZeroDivisionError):
             WorseStruct().__setstate__({}, b'foo')
 
+    @unittest.skipUnless(sys.float_repr_style == 'short',
+                         "applies only when using short float repr style")
```

Comment:
I would prefer to only skip 1e300 test, rather than skipping the whole test method.